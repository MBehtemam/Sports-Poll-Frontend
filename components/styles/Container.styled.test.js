import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";

import Container from "./Container.styled";

describe("Test suits for Container ", () => {
  it("should match with snapshot", () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should has correct style rule on props", () => {
    const tree = mount(<Container direction='row' />);
    expect(tree).toHaveStyleRule("flex-direction", "row");
  });
  it("should has correct style rule with props", () => {
    const tree = mount(<Container />);
    expect(tree).toHaveStyleRule("flex-direction", "column");
  });
});
