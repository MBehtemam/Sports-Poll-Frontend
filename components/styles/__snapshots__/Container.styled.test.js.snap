// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Container  should match with snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  min-height: 100%;
  overflow-y: scroll;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;
